src/mode/mode_normal.ts

```diff
@@ -1,14 +1,27 @@
+import * as _ from 'lodash';
+import * as vscode from 'vscode';
 import {ModeName, Mode} from './mode';
 import {showCmdLine} from './../cmd_line/main';
-import * as vscode from 'vscode';
 
 export default class CommandMode extends Mode {
+    private keyHandler : { [key : string] : (position : vscode.Position) => void; } = {};

```

Comment:
I'm unsure we'll ever need to pass the `position` explicitly. If we need it, we can retrieve it from the current document. Also, why pass just one position when there may be multiple carets?